bittide/src/Bittide/Calendar.hs

```diff
@@ -58,7 +71,7 @@ data ValidEntry a repetitionBits = ValidEntry
   -- 0 = no repetition, entry is valid for one cycle.
   -- 1 = repeated once, entry is valid for two cycles.
   }
-  deriving (BitPack, Eq, Generic, NFDataX, Show, ShowX)
+  deriving (BitPack, Eq, Generic, NFDataX, Show, ShowX, BitPackC, ToFieldType)
```

Comment:
Types that have type arguments need a separate instance for `ToFieldType`, otherwise the arguments will be "inlined" and monomorphised but under the same name. With a separate instance for `ToFieldType` it can generate a Rust type that's also generic!

That said, `repetitionBits` is a `Nat`, that's currently not supported, but I could add support for that in `ToFieldType`. I'll open an issue.